lib/loggingService/LogEntry.js

```diff
@@ -9,8 +9,63 @@ var rootPath = process.cwd();
 var IncomingMessage = require('http').IncomingMessage;
 var ServerResponse = require('http').ServerResponse;
 
+function serializeError(error) {
+	// the stack trace and other error information will be moved into the data object
+
+	// turn the stack string into an array, stripped from its noisy whitespace
+
+	var stack = error.stack && error.stack.split(/\s*\n\s*at\s+/);
+	if (!stack || stack.length <= 1) {
+		return error;
+	}
+
+	// remove the first line: "Error: name", leaving only stack frames
+
+	stack.shift();
+
+	// assign all the data to the data object (custom errors may have some)
+	var data = Object.assign({
+		code: error.code,
+		type: error.name,
+		message: error.message || undefined
+	}, error);
+
+	// reformat the stack
+
+	var locationFound = false;
+
+	for (var i = 0; i < stack.length; i += 1) {
+		// make the file path relative to the project
+
+		stack[i] = stack[i].replace(rootPath + '/', '');
+
+		// If the error did not originate in a native function (like JSON.parse), the top stack
+		// frame should carry file, line and character offset information. While we're looking
+		// at a native function, we move on to the next stack frame to get us closest to the
+		// source of the error in user-land code.
+
+		if (!locationFound) {
+			var m = stack[i].match(/([\w\.]+):([0-9]+):([0-9]+)/);
+			if (m) {
+				locationFound = true;
+
+				data.file = m[1];
+				data.line = parseInt(m[2], 10) || 0;
+				data.offset = parseInt(m[3], 10) || 0;
+			}
+		}
+	}
+
+	data.stack = stack;
```

Comment:
While not technically guaranteed, I make sure to assign stack last because I want it to show as the last attribute when printed on the screen - I find it easier to read.